File: CountingSundays/CS.py
#!/usr/bin/env python3

"""
Project Euler # 19

Brute Force
"""

from datetime import datetime, timedelta

start = datetime(1901, 1, 1)
end = datetime(2000, 12, 31)
count = 0

while start != end:
    # %d - day of the month (i.e. 1st), %w day of the week (i.e. Sunday)
    if start.strftime('%d') == '01' and start.strftime('%w') == '0':
        count += 1
    start += timedelta(days = 1)

print(count)

File: CountingSundays/CS.java
/*
Project Euler # 19

*/

import java.util.Calendar;
import java.util.GregorianCalendar;

public class CS {
   public static void main(String args[]) {
      Calendar start = new GregorianCalendar(1901, Calendar.JANUARY, 1);
      Calendar end = new GregorianCalendar(2000, Calendar.DECEMBER, 31);
      int count = 0;

      while (start.before(end)) {
          if (start.get(Calendar.DAY_OF_WEEK) == Calendar.SUNDAY)
              count++;
          start.add(Calendar.MONTH, 1);
      }
      System.out.println(count);

   }
}
